### PR TITLE
refactor(styles)!: Use "mode" nomenclature instead of "theme"

### PR DESCRIPTION
**Pull request by macandcheese**
BREAKING CHANGE: Use "mode" nomenclature instead of "theme"

- `.calcite-theme-auto`, `.calcite-theme-dark`, and `.calcite-theme-light` CSS classes have been replaced with `.calcite-mode-auto`, `.calcite-mode-dark`, and `.calcite-mode-light `

### CODE REVIEW

**Comment by macandcheese** [comment]
**Related Issue:** #5657 

## Summary
This changes our naming of `theme` to `mode` - meaning css classes change from `.calcite-theme-auto`, `.calcite-theme-dark`, `.calcite-theme-light`, to `.calcite-mode-auto`, `.calcite-mode-dark`, `.calcite-mode-light`.

The concept of a "theme" remains, but it is no longer conflated with "modes". Themes and user overrides can have values associated with both light and dark mode (and, in the future, potentially high contrast, etc.) - but this separation now is more accurate and does a bit of future-proofing.

I think I've made all the needed changes here, there may still be some places where "theme" was used properly as a term (or, in Storybook case, as an import / reference to plugin), and I've tried to leave those alone.

That said, this was a huge search and replace and although all tests are passing, would like a few sets of eyes so opening as a draft. Once it looks good we can add the correct message body for

**Comment by benelan** [comment]
If you are trying to get all of them, there are still references to theme that can be changed to mode in:

- utils/mode.ts
- utils/dom.ts
- input-number.stories.ts
- input.stories.ts
- button.e2e.ts

But none of those are user facing so up to you

File: src/components/dropdown/dropdown.stories.ts [hunk 1]
```diff
@@ -222,12 +222,12 @@ export const darkThemeRTL_TestOnly = (): string => html`
   </calcite-dropdown>
 `;
 
-darkThemeRTL_TestOnly.parameters = { themes: themesDarkDefault };
+darkModeRTL_TestOnly.parameters = { modes: modesDarkDefault };
 
-export const itemsAsLinksDarkTheme = (): string => html`
+export const itemsAsLinksdarkMode = (): string => html`
```
Comment (benelan) [inline src/components/dropdown/dropdown.stories.ts:227]: can you make Dark uppercase here: `itemsAsLinksDarkMode`

File: src/components/dropdown/dropdown.stories.ts [hunk 2]
```diff
@@ -256,7 +256,7 @@ export const itemsAsLinksDarkTheme = (): string => html`
   </calcite-dropdown>
 `;
 
-itemsAsLinksDarkTheme.parameters = { themes: themesDarkDefault };
+itemsAsLinksdarkMode.parameters = { modes: modesDarkDefault };
```
Comment (benelan) [inline src/components/dropdown/dropdown.stories.ts:259]: `itemsAsLinksDarkMode`

File: src/components/tabs/tabs.stories.ts [hunk 3]
```diff
@@ -16,10 +16,10 @@ export default {
   ...storyFilters()
 };
 
-export const simpleDarkThemeRTL_TestOnly = (): string => html`
+export const simpledarkModeRTL_TestOnly = (): string => html`
```
Comment (benelan) [inline src/components/tabs/tabs.stories.ts:19]: `simpleDarkModeRTL_TestOnly`

File: src/components/tabs/tabs.stories.ts [hunk 4]
```diff
@@ -58,14 +58,14 @@ export const bordered = (): string => html`
   </calcite-tabs>
 `;
 
-export const borderedDarkThemeRTL_TestOnly = (): string => html`
+export const bordereddarkModeRTL_TestOnly = (): string => html`
```
Comment (benelan) [inline src/components/tabs/tabs.stories.ts:61]: borderedDarkModeRTL_TestOnly

File: src/components/tile-select/tile-select.stories.ts [hunk 5]
```diff
@@ -121,19 +121,19 @@ export const checkbox_TestOnly = (): string =>
 export const radio_TestOnly = (): string =>
   html`<calcite-tile-select icon="list-radio" heading="test" value="one" type="radio"></calcite-tile-select>`;
 
-export const checkboxDarkThemeRTL_TestOnly = (): string =>
+export const checkboxdarkModeRTL_TestOnly = (): string =>
```
Comment (benelan) [inline src/components/tile-select/tile-select.stories.ts:124]: `checkboxDarkModeRTL_TestOnly`

**Review by benelan — COMMENTED** [review]
A couple story names need a big D, but otherwise the code changes LGTM. Although changing all of the internal variable/method/etc names probably wasn't necessary lol

I'll defer to Matt on final approvals though since this is a breaking change with a big diff

**Comment by driskull** [comment]
This would be a great one to get in before 1.0. 👍 

Let me know when you want me to review.

**Comment by macandcheese** [comment]
@driskull - good to review, had just opened in draft to determine if / when we should make related updates in `calcite-colors`, or otherwise make changes for design tokens, Figma, etc- cc @alisonailea @SkyeSeitz @macandcheese

**Comment by ashetland** [comment]
I can update our Figma kit, but not sure what this means on the tokens side of things.

File: src/components/interfaces.ts [hunk 6]
```diff
@@ -6,6 +6,8 @@ export type FlipContext = "both" | "start" | "end";
 export type Kind = "brand" | "danger" | "info" | "inverse" | "neutral" | "warning" | "success";
 export type Layout = "horizontal" | "vertical" | "grid";
 export type LogicalFlowPosition = "inline-start" | "inline-end" | "block-start" | "block-end";
+export type ModeClass = "calcite-mode-light" | "calcite-mode-dark" | "calcite-mode-auto";
+export type ModeName = "light" | "dark" | "auto";
```
Comment (driskull) [inline src/components/interfaces.ts:10]: Nows the time to decide if "auto" is correct or if we want to make it "device" or something like that 👍

File: src/utils/mode.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { autoMode, darkMode } from "./resources";
+
+/**
+ * Emits when the mode is dynamically toggled between light and dark on <body> or in OS preferences.
+ */
+export function initModeChangeEvent(): void {
+  const { classList } = document.body;
+  const prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
```
Comment (driskull) [inline src/utils/mode.ts:8]: Interesting how matchMedia refers to it as scheme and not mode.

File: src/utils/mode.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  const getTheme = (): string =>
```
Comment (driskull) [inline src/utils/mode.ts:10]: nitpick: The const could be updated to getMode

File: src/utils/dom.spec.ts [hunk 9]
```diff
@@ -301,12 +301,12 @@ describe("dom", () => {
     });
   });
 
-  describe("getThemeName()", () => {
+  describe("getModeName()", () => {
     interface ThemedElement extends HTMLElement {
-      foundThemeName: ThemeName;
+      foundModeName: ModeName;
     }
     function getTestComponentTheme(): string {
```
Comment (driskull) [inline src/utils/dom.spec.ts:308]: nitpick: `getTestComponentMode()`

Comment (driskull) [reply src/utils/dom.spec.ts:308]: I think this file could rename theme in a few more places.

**Review by driskull — APPROVED** [review]
This looks good @macandcheese 👍 💯 

Can we update the `src/demos/theme/auto.html` file path to use mode too?

File: src/components/interfaces.ts (hunk 6 above)
Comment (macandcheese) [reply src/components/interfaces.ts:10]: Good call, will leave this one up to the masses.

Comment (SkyeSeitz) [reply src/components/interfaces.ts:10]: "system"?

Comment (ashetland) [reply src/components/interfaces.ts:10]: "browser"? This mode pulls from the browser setting right?

Comment (macandcheese) [reply src/components/interfaces.ts:10]: Inferred usually from a combination of system / device and browser.

Comment (driskull) [reply src/components/interfaces.ts:10]: Its probably ok to leave it as "auto" since there isn't a perfect word for browser/device/system.

**Comment by macandcheese** [comment]
I think we need a plan to update other instances of "theme" mixin nomenclature, like in `calcite-colors`, and sync with @alisonailea on whether this necessitates any changes in the upcoming token repo / verbiage.

**Pull request merged**